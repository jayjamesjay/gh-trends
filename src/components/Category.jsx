import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { LabelSpan } from '../styles/Span';
import { InputRadio } from '../styles/Input';
import LabelCategory from '../styles/Label';

/**
 * @module Category
 */

/**
 * Single category selector
 *
 * @property {string} label - name of the category
 * @property {boolean} checked - determinates if category is active
 * @property {function} onClick - function fired when category is clicked
 * @returns {Category}
 */
export default function Category({ label, checked, onClick }) {
  const click = useCallback(() => onClick(label), [label, onClick]);

  return (
    <div onClick={click} role="radio" aria-checked={checked} onKeyPress={click} tabIndex="0">
      <LabelCategory active={checked} htmlFor={label}>
        <LabelSpan>{label}</LabelSpan>
      </LabelCategory>
      <InputRadio
        id={label}
        type="radio"
        value={label}
        checked={checked}
        onChange={click}
        tabIndex="-1"
      />
    </div>
  );
}

Category.propTypes = {
  label: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};
