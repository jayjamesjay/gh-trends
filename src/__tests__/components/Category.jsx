import React from 'react';
import 'jest-styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import Category from '../../components/Category';

describe('<Category />', () => {
  const checked = false;
  const label = 'Year';
  const func = () => {};

  it(`renders with default style`, () => {
    const { asFragment } = render(<Category checked={checked} label={label} onClick={func} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`is checked/unchecked`, () => {
    render(<Category checked={checked} label={label} onClick={func} />);
    expect(screen.getByLabelText(label).checked).toEqual(checked);
  });

  it(`contains text of label`, () => {
    render(<Category checked={checked} label={label} onClick={func} />);
    expect(screen.queryByLabelText(label)).not.toBeNull();
  });

  it('fires onClick func', () => {
    let temp = 'Month';
    const onClick = (currLabel) => {
      temp = currLabel;
    };

    render(<Category checked={checked} label={label} onClick={onClick} />);
    fireEvent.click(screen.getByLabelText(label));

    expect(temp).toEqual(label);
  });
});
