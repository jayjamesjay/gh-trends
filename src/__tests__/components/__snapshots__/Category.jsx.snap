// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> renders with default style 1`] = `
<DocumentFragment>
  .c1 {
  position: relative;
  z-index: 100;
}

.c2 {
  opacity: 0;
}

.c0 {
  display: inherit;
  padding: 1rem 1.5rem;
  position: relative;
  background: transparent;
  overflow: hidden;
  font-size: 1.3rem;
  cursor: pointer;
}

.c0::before {
  content: '';
  width: 90%;
  height: 90%;
  position: absolute;
  top: 5%;
  left: 5%;
  z-index: 20;
  background: #fff;
  border-radius: 0.5rem;
  box-shadow: 0px 3px 5px 1px rgba(217,217,217,0.82);
  transition: transform 0.15s ease-in-out;
  transform: translateX(-200%);
}

<div
    aria-checked="false"
    role="radio"
    tabindex="0"
  >
    <label
      class="c0"
      for="Year"
    >
      <span
        class="c1"
      >
        Year
      </span>
    </label>
    <input
      class="c2"
      id="Year"
      tabindex="-1"
      type="radio"
      value="Year"
    />
  </div>
</DocumentFragment>
`;
